src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Tooltip/VSContainerElement.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Tooltip
+{
+    /// <summary>
+    /// Equivalent to VS' ContainerElement. The class has been adapted here so we can
+    /// use it for LSP serialization since we don't have access to the VS version.
+    /// Refer to original class for additional details.
+    /// </summary>
+    internal sealed class VSContainerElement
+    {
+        [JsonProperty("type")]
+        public static readonly string Type = "ContainerElement";
```

Comment:
I had to add a Json "type" property to some classes, for some reason things weren't displaying properly otherwise. Not sure if there's a more "proper" way to do this?

Reply:
You could technically add a JsonConverter for it; however, I think that'd only be important if we didn't own these classes via us re-defining them (we have to because we don't depend on VS). Since we do I think your approach is poifect!

Just for context, the LSP platform [adds the "type"](https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=%2Fsrc%2Fproduct%2FLanguageServer.Protocol.Extensions%2FConverters%2FObjectContentConverter.cs&version=GBdevelop&line=26&lineEnd=35&lineStartColumn=9&lineEndColumn=10&lineStyle=plain&_a=contents) serialized content because all of the `ImageId`, `ImageElement`, `ContainerElement`, `ClassifeidTExtElement`, `ClassifiedTextRun` etc. look too much alike but more importantly it enables them to avoid SumType's because the "type" property exists to say exactly which type to deserialize into 😄